plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -346,8 +346,8 @@ public void registerMLRemoteModel(
         MLTask mlTask,
         ActionListener<MLRegisterModelResponse> listener
     ) {
-        checkAndAddRunningTask(mlTask, maxRegisterTasksPerNode);
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            checkAndAddRunningTask(mlTask, maxRegisterTasksPerNode);
```

Comment:
Can we have a corresponding test to show that exception is 'actionListen'ed now?

Reply:
sure